src/main/java/uk/gov/hmcts/ccd/data/user/DefaultUserRepository.java

```diff
@@ -61,6 +67,13 @@ public IDAMProperties getUserDetails() {
         return restTemplate.exchange(applicationParams.idamUserProfileURL(), HttpMethod.GET, requestEntity, IDAMProperties.class).getBody();
     }
 
+    @Override
```

Comment:
Caching this method for 8 hours, should lend benefits on case events. The call to IDAM was being made just to retrieve the user's first name and last name which then is stored in case event. I don't think user details like first name and last name will change often. What do you think?

Reply:
I think the problem is not so much that it doesn't change often its that when it changes we'll be wrong for a period of time.  Would it make sense to reduce the period to say 1 hour? Or does that lose all the benefits?

Reply:
Yes we could lower it down. Although, lower the time, lesser the benefits. The cache key is the user token, so as long as the user logs out and logs back in (I assume any user detail changes on IDAM will be required to log off and log back in to take effect).

Reply:
Ok for me for caching this info, but I agree we should lower the caching length down a bit.
But we have to be careful here. I don't recall exactly now please @gbenadikar check the documentation. But a TTL of say 1 hour means that **if the cache entry it's not used** within 1 hour, then the entry is evicted. But what happens if that entry is used again and again instead? i.e. if we never reach the TTL? It means we could be returning the outdated IDAM info for days, weeks...To be confirmed

Reply:
Furthermore, if the user was deleted in IDAM, the current code would somehow throw exception I think. Because it wouldn't find the requested user id. 
The new code would allow the case creation to happen anyway. Not a blocker but something to keep in mind

Reply:
@gbenadikar I think my comment above was right, check https://docs.hazelcast.org/docs/latest-dev/manual/html-single/index.html#map
more precisely `max-idle-seconds` which is what we set the ttl config to

Reply:
@mario-paniccia  it is misleading to have TTL set to max idle time on hazelcast cache config. Do you know why we do that? In place of `setMaxIdleSeconds(definitionCacheTTL)`, we should ideally be using `setTimeToLiveSeconds(definitionCacheTTL)`
or atleast renaming the variable definitionCacheTTL

Reply:
@mario-paniccia @dwaynebailey I have lowered down the max idle time for user cache to 4 hours. Bear in mind, the key for this cache is the user token and user id.

Reply:
@mario-paniccia as the cache key is the user token, when a user is deleted/suspended, we should expect the user token to be invalidated by IDAM. The CCD gateway then will block the api call anyway.